File: GlassWeather/Models/Weather.swift
import Foundation

struct Weather: Identifiable {
    let id = UUID()
    let temperature: Double
    let feelsLike: Double
    let humidity: Int
    let pressure: Int
    let windSpeed: Double
    let visibility: Double
    let condition: String
    let description: String
    let sunrise: Date
    let sunset: Date
    let latitude: Double
    let longitude: Double
    
    var isDaytime: Bool {
        let now = Date()
        return now >= sunrise && now < sunset
    }
    
    static let sample = Weather(
        temperature: 22.5,
        feelsLike: 21.0,
        humidity: 65,
        pressure: 1013,
        windSpeed: 3.5,
        visibility: 10000,
        condition: "Partly Cloudy",
        description: "Partly cloudy sky",
        sunrise: Date().addingTimeInterval(-3600),
        sunset: Date().addingTimeInterval(3600 * 8),
        latitude: 35.6762,
        longitude: 139.6503
    )
}

File: web/js/weather.js
// Weather Service
class WeatherService {
    constructor() {
        this.apiKey = 'YOUR_OPENWEATHERMAP_API_KEY';
        this.baseUrl = 'https://api.openweathermap.org/data/2.5';
    }

    async getCurrentWeather(lat, lon) {
        try {
            const response = await fetch(
                `${this.baseUrl}/weather?lat=${lat}&lon=${lon}&appid=${this.apiKey}&units=metric&lang=ja`
            );
            if (!response.ok) throw new Error('天気情報の取得に失敗しました');
            return await response.json();
        } catch (error) {
            throw new Error(`天気API エラー: ${error.message}`);
        }
    }

    async getForecast(lat, lon) {
        try {
            const response = await fetch(
                `${this.baseUrl}/forecast?lat=${lat}&lon=${lon}&appid=${this.apiKey}&units=metric&lang=ja`
            );
            if (!response.ok) throw new Error('予報情報の取得に失敗しました');
            return await response.json();
        } catch (error) {
            throw new Error(`予報API エラー: ${error.message}`);
        }
    }

    async getUVIndex(lat, lon) {
        try {
            const response = await fetch(
                `${this.baseUrl}/uvi?lat=${lat}&lon=${lon}&appid=${this.apiKey}`
            );
            if (!response.ok) throw new Error('UV指数の取得に失敗しました');
            return await response.json();
        } catch (error) {
            throw new Error(`UV API エラー: ${error.message}`);
        }
    }

    convertWeatherData(data) {
        return {
            temp: data.main.temp,
            feelsLike: data.main.feels_like,
            humidity: data.main.humidity,
            pressure: data.main.pressure,
            windSpeed: data.wind.speed,
            windDeg: data.wind.deg,
            visibility: data.visibility,
            condition: data.weather[0].main,
            description: data.weather[0].description,
            icon: data.weather[0].icon,
            clouds: data.clouds.all,
            sunset: data.sys.sunset,
            sunrise: data.sys.sunrise
        };
    }
}
